gatsby/content/docs/free-small-matrix-server.mdx

```diff
@@ -0,0 +1,204 @@
+
+# Free Small Matrix Server
+
+Ingredients:
+
+- Domain
+- Server
+- Software
+- Configuration
+
+## Free domain name
+
+Get one from [Freenom] for free or any other place you wish. Freenom only gives
+free domains for a year. A Matrix server cannot be migrated (yet?) from one
+domain to another, so you might want to get a longer-lived one.
+
+- https://tld-list.com/ - find cheap domains, look at renewal fees also
+- https://www.namecheap.com/promos/amazing98s/ - cheap first year
+
+You also need to specify subdomains (which is why most dynamic dns services
+aren't sufficient). To do this I added the domain on the free [Cloudflare] plan.
+You can also use freenom dns management if you prefer.
+
+[Freenom]: https://www.freenom.com
+[Cloudflare]: https://www.cloudflare.com
+
+## Get a free server
+
+Comparison of free cloud offerings (2019/10):
+
+| Vendor   | Time-limit | Count                    | RAM (GB) | Storage (GB) | Transfer (GB) |
+| -------- | ---------- | ------------------------ | -------: | -----------: | ------------: |
+| [AWS]    | 12 months  | 1 t2.micro               | 1        | 30           | 15            |
+| [Azure]  | 12 months  | 1 B1S                    | 1        | 2x 64        | 15            |
+| [GCP]    | no limit   | 1 f1-micro               | 0.6      | 30           | 1             |
+| [Oracle] | no limit   | 2 VM.Standard.E2.1.Micro | 1        | 100          | 10000         |
+
+[AWS]: https://aws.amazon.com/free/
+[Azure]: https://azure.microsoft.com/en-us/free/
+[GCP]: https://cloud.google.com/free/
+[Oracle]: https://www.oracle.com/cloud/free/
+
+If you choose to, register on Oracle Cloud (requires credit card for
+verification) and create a free VM:
+
+1. Create a VM instance
+2. Give it a name
+3. OS/image: Canonical Ubuntu 18.04 Minimal
+4. Show Shape, Network and Storage Options -> Assign a public IP address
+5. Add SSH key ([generate one][sshkey] using `ssh-keygen` if you don't have one)
+6. Show advanced options. I removed monitoring, since I'll remove the monitoring
+   agent later.
+
+View resources and make sure the instance and boot volume are "Always Free" if
+you don't intent to pay for them.
+
+Make sure you can ssh to the server (using the ssh key you generated/picked)
+using the user 'ubuntu' and can use `sudo`.
+
+[sshkey]: https://docs.oracle.com/en/cloud/iaas/compute-iaas-cloud/stcsg/generating-ssh-key-pair.html
+
+## Open firewall
+
+### Oracle Cloud Security lists
+
+View resources -> Instances -> Select instance -> Virtual Cloud Network ->
+Public Subnet -> Security Lists -> Default -> Ingress
+
+Open incoming for CIDR 0.0.0.0/0:
+- 22/tcp for SSH (should be open already)
+- 80/tcp for HTTP
+- 443/tcp for HTTPS
+- 8448/tcp for Matrix federation
+- 3478/tcp, 5349/tcp, 3478/udp, 5349/udp, 49152-49172/udp for TURN/STUN (you can
+  omit this if you don't run coturn for audio and video calls)
```

Comment:
New users will likely not know what this means. I might reword to:

```suggestion
  omit this if you don't wish to support 1-to-1 audio and video calls)
```

Reply:
I've been thinking about this wording, since it's not actually needed for calls, but usually helps if there are firewalls or NATs

Reply:
hope you like this wording better

Reply:
While it isn't *strictly required* for calls, it is basically required if you want calls to work reliably so personally I prefer anoa's wording.

Reply:
Ok, I just don't want to give the wrong impression of it being needed. Lots of places that don't care about calls, just need an alternative to IRC/Slack. I tried to cover this in one place